src/pages/constitution/Constitution.tsx

```diff
@@ -13,61 +13,61 @@ import {
   IonTitle,
   IonToolbar,
   IonIcon,
-  IonButton
+  IonButton,
+  withIonLifeCycle
 } from '@ionic/react';
 import './Constitution.css';
-import { useParams } from 'react-router-dom';
+import { RouteComponentProps } from 'react-router-dom';
 import { arrowBack } from 'ionicons/icons';
-import { useRecoilState } from 'recoil';
-import { constitutionState } from '../../recoil/atoms/constitution';
-import * as data from '../../assets/data/constitution.json';
+import * as constitution from '../../assets/data/constitution.json';
 
-const Tab1: React.FC = () => {
-  const provisionRef: any = React.createRef();
-  const [constitution, setConstitution] = useRecoilState(constitutionState);
+function previous() {
+  window.history.back();
+}
 
-  useEffect(() => {
-    console.log('setting data')
-    setConstitution({
-      toc: data.toc,
-      body: data.body
-    });
-  }, [constitution.toc.length, setConstitution]);
+interface Props extends RouteComponentProps<{ id: string; }> { }
 
-  const { id } = useParams()
+class Constitution extends React.Component<Props> {
+  private readonly rootRef: React.RefObject<HTMLDivElement>;
+  private readonly constitutionRoot: Element | null;
+  
+  constructor(props: any) {
+    super(props);
+    this.rootRef = React.createRef();
 
-  useEffect(() => {
-    if (provisionRef.current) {
-      provisionRef.current.innerHTML = constitution.body;
-    }
-  }, [constitution, provisionRef]);
-
-  useEffect(() => {
-    if (id) {
-      setTimeout(() => {
-        scroll(id);
-      }, 800);
-    }
-  })
+    // parse the constitution HTML once
+    this.constitutionRoot = new DOMParser().parseFromString(
```

Comment:
This is crucial -- it's a huge amount of HTML and it was being parsed every time the FC render function was called.